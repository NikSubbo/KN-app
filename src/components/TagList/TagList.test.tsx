import { render } from '@testing-library/react';
import Taglist, { Props } from './TagList';

const DEFAULT_PROPS = {
  tags: ['PDF', 'Change', 'Create', 'Maintenance', 'Business', 'FoxIT']
};

const renderContainer = (overrideProps?: Partial<Props>) => {
  return render(<Taglist {...DEFAULT_PROPS} {...overrideProps} />);
};

describe('TagList', () => {
  it('should match the snapshot', () => {
    const { container } = renderContainer();
    expect(container).toMatchSnapshot();
  });

  it('should return null for the empty tags array', () => {
    const { container } = renderContainer({
      tags: []
    });
    expect(container.firstChild).toBeNull();
  });
});
