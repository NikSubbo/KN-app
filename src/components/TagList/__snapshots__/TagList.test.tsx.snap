// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TagList should match the snapshot 1`] = `
<div>
  <div
    class="flex flex-wrap"
  >
    <div
      class="text-xs font-medium text-aqua px-2 py-1 rounded bg-aquaLight mr-2 mt-2"
    >
      PDF
    </div>
    <div
      class="text-xs font-medium text-aqua px-2 py-1 rounded bg-aquaLight mr-2 mt-2"
    >
      Change
    </div>
    <div
      class="text-xs font-medium text-aqua px-2 py-1 rounded bg-aquaLight mr-2 mt-2"
    >
      Create
    </div>
    <div
      class="text-xs font-medium text-aqua px-2 py-1 rounded bg-aquaLight mr-2 mt-2"
    >
      Maintenance
    </div>
    <div
      class="text-xs font-medium text-aqua px-2 py-1 rounded bg-aquaLight mr-2 mt-2"
    >
      Business
    </div>
    <div
      class="text-xs font-medium text-aqua px-2 py-1 rounded bg-aquaLight mr-2 mt-2"
    >
      FoxIT
    </div>
  </div>
</div>
`;
